compiler/x/codegen/OMRTreeEvaluator.cpp

```diff
@@ -2560,34 +2566,16 @@ TR::Register *OMR::X86::TreeEvaluator::OverflowCHKEvaluator(TR::Node *node, TR::
       }
 
    cg->setVMThreadRequired(true);
-   //the BBStartEvaluator will generate the label but in this case the catch block might not been evaluated yet
-   TR::Node * bbstartNode = overflowCatchBlock->getEntry()->getNode();
-   TR_ASSERT((bbstartNode->getOpCodeValue() == TR::BBStart), "catch block entry %p must be TR::BBStart\n", bbstartNode);
-   if (!bbstartNode->getLabel()) 
-      {
-      TR::LabelSymbol *label = generateLabelSymbol(cg);
-      bbstartNode->setLabel(label);
-      }
-
-   bool isUnsigned = false;
-   switch (node->getOverflowCheckOperation())
-   {
-       case TR::buadd:
-       case TR::cadd:
-       case TR::iuadd:
-       case TR::luadd:
-       case TR::busub:
-       case TR::csub:
-       case TR::iusub:
-       case TR::lusub:
-           isUnsigned = true;
-           break;
    }
 
-   TR_X86OpCodes opcode = isUnsigned? JB4: JO4; 
+TR::Register *OMR::X86::TreeEvaluator::overflowCHKEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   TR::Block *overflowCatchBlock = TR::TreeEvaluator::getOverflowCatchBlock(node, cg);
+   TR::TreeEvaluator::genArithmeticInstructionsForOverflowCHK(node, cg);
+   TR_X86OpCodes opcode = node->getOpCodeValue() == TR::OverflowCHK? JO4: JB4;
```

Comment:
I would prefer this line to be expanded for the two OverflowCHK cases we have now along with a TR_ASSERT if an opcode other than those two go through here, as some small amount of future proofing. i.e. something like:
```
if (node->getOpCodeValue() == TR::OverflowCHK)
   opcode = JO4;
else if (node->getOpCodeValue() == TR::UnsignedOverflowCHK)
   opcode = JB4;
else
   TR_ASSERT(0, "unrecognized overflow operation in overflowCHKEvaluator");
```